packages/expo-image/src/web/ImageWrapper.tsx

```diff
@@ -24,6 +24,73 @@ function getImgPropsFromSource(source: ImageSource | SrcSetSource | null | undef
   return {};
 }
 
+function useThumbhash(source: ImageSource | null | undefined) {
+  const isThumbhash = isThumbhashString(source?.uri || '');
+  const strippedThumbhashString = source?.uri?.replace(/thumbhash:\//, '') ?? '';
+  const thumbhashSource = useMemo(
+    () => (isThumbhash ? { uri: thumbHashStringToDataURL(strippedThumbhashString) } : null),
+    [strippedThumbhashString, isThumbhash]
+  );
+  return thumbhashSource;
+}
+
+function useImageHashes(source: ImageSource | null | undefined) {
+  const thumbhash = useThumbhash(source);
+  const blurhash = useBlurhash(source);
+  return useMemo(
+    () => ({
+      resolvedSource: blurhash ?? thumbhash ?? source,
+      isImageHash: !!blurhash || !!thumbhash,
+    }),
+    [blurhash, thumbhash]
+  );
+}
+
+function useHeaders(
+  source: ImageSource | null | undefined,
+  cachePolicy: ImageNativeProps['cachePolicy'],
+  onError?: OnErrorEvent[]
+): ImageSource | null | undefined {
+  const [objectURL, setObjectURL] = useState<string | null>(null);
+  useEffect(() => {
+    (async () => {
+      if (!source?.headers || !source.uri) {
+        return;
+      }
+      try {
+        const result = await fetch(source.uri, {
+          headers: source.headers,
+          cache: cachePolicy === 'none' ? 'no-cache' : 'default',
+          redirect: 'follow',
+        });
+        if (!result.ok) {
+          throw new Error(`Failed to fetch image: ${result.status} ${result.statusText}`);
+        }
+        const blob = await result.blob();
+        setObjectURL(URL.createObjectURL(blob));
+      } catch (error) {
+        console.error(error);
```

Comment:
```suggestion
```

Reply:
Users can receive that error using the callback, so I don't see why we should log it. Also, it will be present in the production version, which may not be a desired behavior.